### PR TITLE
[in_app_purchase_android] Introduced new ReplacementMode for Android's billing client

### PR DESCRIPTION
**Pull request by vongrad**
Introduced new `ReplacementMode` for Android's billing client and deprecated `ProrationMode`.

This PR is a follow-up on [https://github.com/flutter/packages/pull/6403](https://github.com/flutter/packages/pull/6403), where it was decided that we should not replace the `ProrationMode` with `ReplacementMode`, but instead only deprecate `ProrationMode`. The reason for a new PR is also that `in_app_purchase_android` version `0.3.3` changed internal platform communication to Pigeon, which meant I had to make major changes to my original PR.

*List which issues are fixed by this PR. You must list at least one issue.*
https://github.com/flutter/flutter/issues/128957

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by vongrad** [comment]
@bparrishMines I have made this PR in favor of https://github.com/flutter/packages/pull/6403#issuecomment-2023552466. Any chance it can be reviewed?

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.3.4
+
+* Introduces new ReplacementMode for Android's billing client
```
Comment (bparrishMines) [inline packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md:3]: nit:

```suggestion
* Introduces new `ReplacementMode` for Android's billing client.
```

Comment (bparrishMines) [inline packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md:3]: This should also add that `ProrationMode` is being deprecated.

**Review by bparrishMines — CHANGES_REQUESTED** [review]
LGTM

@gmackall could you give this a secondary review

**Review by gmackall — APPROVED** [review]
LGTM minus the nits @bparrishMines requested

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md (hunk 1 above)
Comment (vongrad) [reply packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md]: I made the requested changes

**Comment by vongrad** [comment]
I made the requested changes. I have however made a mistake with rebasing the branch with `main` and pushing, hence all the new required reviewers. I rolled it back, but the extra reviewers remained on the PR,  any chance that you can remove them?

**Comment by vongrad** [comment]
Anything missing from my end?

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +14,7 @@
 
 * Adds `countryCode` API.
 
+
```
Comment (domesticmouse) [inline packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md:17]: nit: delete this line

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md (hunk 2 above)
Comment (vongrad) [reply packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md]: The line was removed

**Review by domesticmouse — APPROVED** [review]

**Comment by vongrad** [comment]
@bparrishMines I have made the changes you requested, could you review it?

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**